pkg/freeze/containerd/pause.go

```diff
@@ -35,51 +45,57 @@ func New() (*Containerd, error) {
 	}
 
 	return &Containerd{
-		conn: conn,
+		conn:       conn,
+		containerd: c,
 	}, nil
 }
 
 // Freeze freezes the user container via the freezer cgroup.
-func (f *Containerd) Freeze(ctx context.Context, podName, containerName string) error {
+func (f *Containerd) Freeze(ctx context.Context, podName string) error {
 	ctrd, err := containerd.NewWithConn(f.conn)
 	if err != nil {
 		return err
 	}
 
-	containerID, err := lookupContainerID(ctx, f.conn, podName, containerName)
+	containers, err := f.containerd.List(ctx, f.conn, podName)
+	containerIDs, err := lookupContainerIDs(containers)
 	if err != nil {
 		return err
 	}
 
-	ctx = namespaces.WithNamespace(ctx, "k8s.io")
-	if _, err := ctrd.TaskService().Pause(ctx, &tasks.PauseTaskRequest{ContainerID: containerID}); err != nil {
-		return err
+	for _, c := range containerIDs {
+		if err := f.containerd.Pause(ctx, ctrd, c); err != nil {
+			return fmt.Errorf("%s not paused: %v", c, err)
+		}
 	}
-
 	return nil
 }
 
 // Thaw thats a container which was freezed via the Freeze method.
-func (f *Containerd) Thaw(ctx context.Context, podName, containerName string) error {
+func (f *Containerd) Thaw(ctx context.Context, podName string) error {
 	ctrd, err := containerd.NewWithConn(f.conn)
 	if err != nil {
 		return err
 	}
 
-	containerID, err := lookupContainerID(ctx, f.conn, podName, containerName)
+	containers, err := f.containerd.List(ctx, f.conn, podName)
+	containerIDs, err := lookupContainerIDs(containers)
 	if err != nil {
 		return err
 	}
 
-	ctx = namespaces.WithNamespace(ctx, "k8s.io")
-	if _, err := ctrd.TaskService().Resume(ctx, &tasks.ResumeTaskRequest{ContainerID: containerID}); err != nil {
-		return err
+	for _, c := range containerIDs {
+		if err := f.containerd.Resume(ctx, ctrd, c); err != nil {
+			return fmt.Errorf("%s not resumed: %v", c, err)
+		}
 	}
-
 	return nil
 }
 
-func lookupContainerID(ctx context.Context, conn *grpc.ClientConn, podUID, containerName string) (string, error) {
+type ContainerdCri struct{}
```

Comment:
```suggestion
type ContainerdCRI struct{}
```

(go acronyms should be consistent case)